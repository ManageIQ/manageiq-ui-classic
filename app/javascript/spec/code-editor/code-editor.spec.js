import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CodeEditor, { DataDrivenFormCodeEditor } from '../../components/code-editor';

jest.mock('@@ddf', () => ({
  useFieldApi: props => ({ meta: {}, input: {}, ...props }),
}));

describe('CodeEditor component', () => {
  let initialProps;
  beforeEach(() => {
    initialProps = {
      onChange: jest.fn(),
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<CodeEditor {...initialProps} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render mode switches', () => {
    const wrapper = shallow(<CodeEditor {...initialProps} modes={['yaml', 'json']} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should mount and assign correct props to data driven variant', () => {
    const onChange = jest.fn();
    const wrapper = mount(<DataDrivenFormCodeEditor {...initialProps} onChange={onChange} />);
    wrapper.find(CodeEditor).props().onChange('foo');
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith('foo');
  });
});
