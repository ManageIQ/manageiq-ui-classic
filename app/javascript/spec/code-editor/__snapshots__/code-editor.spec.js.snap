// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeEditor component should render correctly 1`] = `
<FormGroup
  hasMargin={true}
  invalid={false}
  legendText="foo"
  message={false}
  messageText=""
>
  <Controlled
    className="miq-codemirror "
    editorDidMount={[Function]}
    onBeforeChange={[Function]}
    onChange={[MockFunction]}
    options={
      Object {
        "autoCloseBrackets": true,
        "gutters": Array [
          "CodeMirror-lint-markers",
        ],
        "lineNumbers": true,
        "lineWrapping": true,
        "lint": true,
        "mode": "yaml",
        "styleActiveLine": true,
        "theme": "eclipse",
      }
    }
    style={
      Object {
        "height": "auto",
      }
    }
  />
  <HelperTextBlock />
</FormGroup>
`;

exports[`CodeEditor component should render mode switches 1`] = `
<FormGroup
  hasMargin={true}
  invalid={false}
  legendText="foo"
  message={false}
  messageText=""
>
  <RadioButtonGroup
    labelPosition="right"
    name="--undefined--mode"
    onChange={[Function]}
    orientation="horizontal"
    valueSelected="yaml"
  >
    <ForwardRef(RadioButton)
      key="yaml"
      labelText="yaml"
      value="yaml"
    />
    <ForwardRef(RadioButton)
      key="json"
      labelText="json"
      value="json"
    />
  </RadioButtonGroup>
  <Controlled
    className="miq-codemirror "
    editorDidMount={[Function]}
    onBeforeChange={[Function]}
    onChange={[MockFunction]}
    options={
      Object {
        "autoCloseBrackets": true,
        "gutters": Array [
          "CodeMirror-lint-markers",
        ],
        "lineNumbers": true,
        "lineWrapping": true,
        "lint": true,
        "mode": "yaml",
        "styleActiveLine": true,
        "theme": "eclipse",
      }
    }
    style={
      Object {
        "height": "auto",
      }
    }
  />
  <HelperTextBlock />
</FormGroup>
`;
